lms/static/scripts/postmessage_json_rpc/server/index.js

```diff
@@ -1,15 +1,32 @@
 import Server from './server';
-import { requestConfig, requestGroups } from './methods';
+import { requestConfig } from './methods';
 
 let server = {}; // Singleton RPC server reference
 
 /**
- * Create a new RPC server and pass in a the requestConfig object
+ * Create a new RPC server and register any methods it will support.
+ *
+ * @return {Server} - Instance of the server.
  */
 function startRpcServer() {
   server = new Server();
   server.register('requestConfig', requestConfig);
-  server.register('requestGroups', requestGroups);
+
+  /**
+   * In the case where groups are not available at load time, the groups must
+   * be fetched asynchronously using the api values found in the js-config
+   * object. In order to speed this up even more, the groups are pre-loaded
+   * before this RPC endpoint (`requestGroups`) is queried -- This method
+   * simply returns an awaited promise to the client.
+   */
+  const groupsPromise = new Promise(resolve => {
+    server.resolveGroupFetch = resolve;
+  });
+  server.register('requestGroups', async () => {
+    return await groupsPromise;
+  });
```

Comment:
I can't see a test for this. Can you add one?

Reply:
done